src/libs/actions/Session/index.js

```diff
@@ -345,7 +345,16 @@ function signIn(password, validateCode, twoFactorAuthCode, preferredLocale = CON
     API.write('SigninUser', params, {optimisticData, successData, failureData});
 }
 
-function signInWithValidateCode(accountID, validateCode, twoFactorAuthCode) {
+function signInWithValidateCode(accountID, code, twoFactorAuthCode) {
+    // If this is called from the 2fa step, get the validateCode directly from onyx
+    // instead of the one passed from the component state because the state is changing when this method is called.
+    let validateCode;
+    if (twoFactorAuthCode) {
+        validateCode = credentials.validateCode;
+    } else {
+        validateCode = code;
+    }
```

Comment:
NAB but since this isn't being overwritten later
```suggestion
const validateCode = twoFactorAuthCode ? credentials.validateCode : code;
```